File: 4126-MinimumNumberOfFlipsToReverseBinaryString/4126-MinimumNumberOfFlipsToReverseBinaryString.cpp
// Last updated: 7/21/2026, 11:49:17 AM
class Solution {
public:
    int minimumFlips(int n) {
        string s = decToBin(n);
        int i = 0;
        int j = s.length() - 1;
        int c = 0;

        while (i <= j) {
            if (s[i] != s[j]) c += 2;
            i++;
            j--;
        }
        return c;
    }

    string decToBin(int n) {
        string s = "";
        while (n > 0) {
            int rem = n % 2;
            s += (rem + '0');   // convert int to char
            n = n / 2;
        }
        reverse(s.begin(), s.end());
        return s;
    }
};

File: 4126-MinimumNumberOfFlipsToReverseBinaryString/4126-MinimumNumberOfFlipsToReverseBinaryString.java
// Last updated: 7/21/2026, 11:49:25 AM
class Solution {
    public int minimumFlips(int n) {
        String s=decToBin(n);
        int i=0;
        int j=s.length()-1;
        int c=0;
        while(i<=j){
            if(s.charAt(i)!=s.charAt(j)) c+=2;
            i++;
            j--;
        }
        return c;
    }
    public static String decToBin(int n){
        StringBuilder sb=new StringBuilder();
		while(n>0) {
			int rem=n%2;
            sb.append(rem);
			n=n/2;
		}
		return sb.reverse().toString();
    }
}